lib/main.dart

```diff
@@ -1,20 +1,28 @@
+import 'dart:async';
+
 import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
 import 'package:flutter/material.dart';
 import 'package:presentation/presentation.dart';
 import 'package:star_movie/di/init_dependencies.dart';
 import 'package:star_movie/flavor_configurator/flavor_configurator.dart';
 import 'firebase_options.dart';
 
 Future<void> initApp(Flavor flavor) async {
-  WidgetsFlutterBinding.ensureInitialized();
-  final flavorConfigurator = FlavorConfigurator(flavor: flavor);
-  await Firebase.initializeApp(
-    options: DefaultFirebaseOptions.currentPlatform,
-  );
-  await initDependencies(flavorConfigurator);
-  runApp(
-    StarMovieApp(
-      config: await flavorConfigurator.configurePresentation(),
-    ),
-  );
+  runZonedGuarded<Future<void>>(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    final flavorConfigurator = FlavorConfigurator(flavor: flavor);
+    await Firebase.initializeApp(
+      options: DefaultFirebaseOptions.currentPlatform,
+    );
+    FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterFatalError;
+    await initDependencies(flavorConfigurator);
+    runApp(
+      StarMovieApp(
+        config: await flavorConfigurator.configurePresentation(),
+      ),
+    );
+  },
+      (error, stack) =>
+          FirebaseCrashlytics.instance.recordError(error, stack, fatal: true));
```

Comment:
is that right?Because the documentation says:
void main() async {
  runZonedGuarded<Future<void>>(() async {
    WidgetsFlutterBinding.ensureInitialized();
    await Firebase.initializeApp();
    // The following lines are the same as previously explained in "Handling uncaught errors"
    FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterError;

    runApp(MyApp());
  }, (error, stack) => FirebaseCrashlytics.instance.recordError(error, stack));
}

Reply:
yes, it's work well, cuz main function persists in flavour files:
Future<void> main() async => initApp(Flavor.prod);

Reply:
okay,thanks

Reply:
correct me if I'm wrong, but as I understand, crashlytics works both for debug and prod, while we agreed to add it only to prod

Reply:
I think we can even avoid some code and do like this

<img width="589" alt="image" src="https://user-images.githubusercontent.com/64923802/195835229-bc4a9042-551e-4ce4-a533-8f3a1b42841f.png">

but need to check if it works ok

Reply:
tested, it should work)